collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java

```diff
@@ -19,70 +19,57 @@
+import timber.log.Timber;
 
 public class ApplicationInitializer {
 
     private final Application context;
     private final UserAgentProvider userAgentProvider;
-    private final SettingsPreferenceMigrator preferenceMigrator;
     private final PropertyManager propertyManager;
     private final Analytics analytics;
-    private final Settings generalSettings;
-    private final Settings adminSettings;
-    private final Settings metaSettings;
     private final StorageInitializer storageInitializer;
-    private final AppStateProvider appStateProvider;
-    private final ProjectImporter projectImporter;
-    private final CurrentProjectProvider currentProjectProvider;
-
-    public ApplicationInitializer(Application context, UserAgentProvider userAgentProvider, SettingsPreferenceMigrator preferenceMigrator,
-                                  PropertyManager propertyManager, Analytics analytics, StorageInitializer storageInitializer, Settings generalSettings,
-                                  Settings adminSettings, Settings metaSettings, AppStateProvider appStateProvider, ProjectImporter projectImporter, CurrentProjectProvider currentProjectProvider) {
+    private final LaunchState launchState;
+    private final AppUpgrader appUpgrader;
+
+    @Inject
+    public ApplicationInitializer(Application context, UserAgentProvider userAgentProvider,
+                                  PropertyManager propertyManager, Analytics analytics,
+                                  StorageInitializer storageInitializer, LaunchState launchState,
+                                  AppUpgrader appUpgrader) {
         this.context = context;
         this.userAgentProvider = userAgentProvider;
-        this.preferenceMigrator = preferenceMigrator;
         this.propertyManager = propertyManager;
         this.analytics = analytics;
-        this.generalSettings = generalSettings;
-        this.adminSettings = adminSettings;
-        this.metaSettings = metaSettings;
         this.storageInitializer = storageInitializer;
-        this.appStateProvider = appStateProvider;
-        this.projectImporter = projectImporter;
-        this.currentProjectProvider = currentProjectProvider;
+        this.launchState = launchState;
+        this.appUpgrader = appUpgrader;
     }
 
     public void initialize() {
         initializeStorage();
-        initializePreferences();
+        performUpgradeIfNeeded();
         initializeFrameworks();
         initializeLocale();
-        importExistingProjectIfNeeded();
+
+        launchState.appLaunched();
     }
 
-    private void initializeStorage() {
-        storageInitializer.createOdkDirsOnStorage();
+    private void performUpgradeIfNeeded() {
+        if (launchState.isUpgradedFirstLaunch()) {
+            appUpgrader.upgrade();
+        }
     }
 
-    private void initializePreferences() {
-        performMigrations();
-        reloadSharedPreferences();
```

Comment:
There is no test to drive this out and I think it might just be a relic at this point? Our `SharedPreferencesSettings` returns defaults if values are missing, so there is no reason to write them to the prefs themselves (as far as I understand).

Reply:
I could just delete it as part of this PR I guess?

Reply:
Actually there are cases where loading defaults is important... ah that's confusing!
For example take a look at `SettingsImporter` where we load defaults after importing new settings (we don't use that `setDefaultForAllSettingsWithoutValues()` method, maybe we should to remove duplicated code but generally it does the same)
We need loading defaults because next step it to set up `SettingsChangeHandler` for every setting so that it can perform operations after changing some settings not only for those preferences that were included in json but for all settings (those just cleared as well - taht's why we need to save defaults to loop through them as well).
I'm not sure if in our code base we have settings that really need to react to clearing them but at least theoretically it should work like that.

Reply:
Hmmm we can come back to this later then! I think hopefully we can get away with just using the default behaviour in `SharedPreferencesSettings` eventually.

Reply:
So maybe we should have an issue for investigating it - when we need loading defaults?